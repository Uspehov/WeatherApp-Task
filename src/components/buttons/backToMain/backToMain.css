.backToMain-button {
    height: 10%;
    background-color: cornflowerblue;
    text-align: center;
    font-size: 23px;
    margin: auto;
    margin-top: 40px;
    margin-bottom: 40px;
}