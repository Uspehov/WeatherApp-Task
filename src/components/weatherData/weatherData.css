.weatherData-container {
    font-size: 25px; 
}

.weatherData-subtitle {
    margin: 15px;
}

.weatherData-columnContainer {
    margin: auto;
    width: 35%;
    display: flex;
    align-items: center;
    justify-content: space-between;
    margin-top: 40px;
}

.weatherData-column {
    display: flex;
    flex-direction: column;
}

@media (max-width: 750px) {
    .weatherData-columnContainer {width: 65%;}
}
