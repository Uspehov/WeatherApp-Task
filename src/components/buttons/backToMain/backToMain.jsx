import React from 'react';
import { Link } from "react-router-dom";
import './backToMain.css';

export function BackToMainButton() {
    return (
        <Link replace={true} to={`/`} className={`backToMain-button mainButton`}>Go Back</Link>
    )
}
