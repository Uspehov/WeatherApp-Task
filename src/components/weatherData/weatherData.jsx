import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { BackToMainButton } from '../buttons/backToMain/backToMain';
import { weatherValues } from '../../js/constants/weatherValues';
import './weatherData.css';

export class WeatherData extends Component {

    render() {
        const data = this.props.data;
        return (
            <div className='weatherData-container'>
                <h1>{this.props.cityName}</h1>
                <div >
                    <h3 className='weatherData-subtitle'>{data.weather[0].main}</h3>
                    <span>{data.weather[0].description}</span>
                    <div className='weatherData-columnContainer'>
                        <div className='weatherData-column'>{Object.keys(data.main).map((item) => <div key={item}>{item.charAt(0).toLocaleUpperCase() + item.slice(1).replace(/_/, '-')}</div>)}</div>
                        <div className='weatherData-column'>{Object.values(data.main).map((item, index) => <div key={item}>{item} {weatherValues[index]}</div>)} </div>
                    </div>
                    <h3 className='weatherData-subtitle'>Wind</h3>
                    <div className='weatherData-columnContainer'>
                        <div className='weatherData-column'>{Object.keys(data.wind)[0].charAt(0).toLocaleUpperCase() + Object.keys(data.wind)[0].slice(1)}</div>
                        <div className='weatherData-column'>{Object.values(data.wind)[0]} m/s</div>
                    </div>
                </div>
                <BackToMainButton />
            </div>
        );
    }
}

WeatherData.propTypes = {
    cityName: PropTypes.string,
    data: PropTypes.object
}