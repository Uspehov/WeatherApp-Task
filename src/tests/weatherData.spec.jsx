import React from 'react';
import { mount, configure } from 'enzyme';
import { WeatherData } from '../components/weatherData/weatherData';
import { BackToMainButton } from "../components/buttons/backToMain/backToMain";
import { BrowserRouter as Router } from 'react-router-dom';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Weather Data Page', () => {
    const weatherDataProps = {
        cityName: 'Moscow',
        data: {
            weather: [{
                data: 'sun',
                main: 'clear',
                wind: '4',
                description: 'good day'
            }],
            main: {
                humidity: 79,
                pressure: 1030,
                temp: 268.03,
                temp_max: 269.26,
                temp_min: 267.04
            },
            wind: {speed: 2, deg: 170}
        }
    };

    it('should render', () => {
        const wrapper = mount(<Router><WeatherData {...weatherDataProps} /></Router>);
        expect(wrapper.debug()).toMatchSnapshot();
    });

    it('back to main button should be presented', () => {
        const wrapper = mount(<Router><WeatherData {...weatherDataProps} /></Router>);
        expect(wrapper.find(BackToMainButton).length).toBe(1);
    });
});