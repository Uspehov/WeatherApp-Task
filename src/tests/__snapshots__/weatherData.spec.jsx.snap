// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weather Data Page should render 1`] = `
"<BrowserRouter>
  <Router history={{...}}>
    <WeatherData cityName=\\"Moscow\\" data={{...}}>
      <div className=\\"weatherData-container\\">
        <h1>
          Moscow
        </h1>
        <div>
          <h3 className=\\"weatherData-subtitle\\">
            clear
          </h3>
          <span>
            good day
          </span>
          <div className=\\"weatherData-columnContainer\\">
            <div className=\\"weatherData-column\\">
              <div>
                Humidity
              </div>
              <div>
                Pressure
              </div>
              <div>
                Temp
              </div>
              <div>
                Temp-max
              </div>
              <div>
                Temp-min
              </div>
            </div>
            <div className=\\"weatherData-column\\">
              <div>
                79
                 
                °F
              </div>
              <div>
                1030
                 
                Pa
              </div>
              <div>
                268.03
                 
                %
              </div>
              <div>
                269.26
                 
                °F
              </div>
              <div>
                267.04
                 
                °F
              </div>
               
            </div>
          </div>
          <h3 className=\\"weatherData-subtitle\\">
            Wind
          </h3>
          <div className=\\"weatherData-columnContainer\\">
            <div className=\\"weatherData-column\\">
              Speed
            </div>
            <div className=\\"weatherData-column\\">
              2
               m/s
            </div>
          </div>
        </div>
        <BackToMainButton>
          <Link replace={true} to=\\"/\\" className=\\"backToMain-button mainButton\\">
            <LinkAnchor className=\\"backToMain-button mainButton\\" href=\\"/\\" navigate={[Function: navigate]}>
              <a className=\\"backToMain-button mainButton\\" href=\\"/\\" onClick={[Function: onClick]}>
                Go Back
              </a>
            </LinkAnchor>
          </Link>
        </BackToMainButton>
      </div>
    </WeatherData>
  </Router>
</BrowserRouter>"
`;
